Cargo.lock

```diff
@@ -11235,7 +11235,7 @@ dependencies = [
  "sc-network-sync",
  "sc-service",
  "sc-transaction-pool-api",
- "sp-runtime",
+ "sp-runtime 7.0.0 (git+https://github.com/subspace/substrate?rev=9cf78129a2638d3f370868863d16f4fe32b4ad30)",
```

Comment:
Where did this change come from? It indicates two versions of Substrate are mixed in the same repo, which should not happen and must be fixed.

Reply:
This came from fraud proof pulling a different revision for some reason (so we were compiling large chunk of Substrate twice. Fixed in latest commit.

@liuchengxu and @NingLin-P please keep an eye on lock file when this happens again, there should always be exactly one version of Substrate, none of dependencies should have source added like this explicitly.

Reply:
I had this issue before in my local environment but it supposes to fail to compile due to type mismatch (in Rust, libraries from different sources have different type id I guess), the reason `subspace-fraud-proof` can be compiled is `frame-support` is actually an unused dependency, adding an unused dependency check to our CI may help this situation.